File: test_login.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page()
    page.goto('http://localhost:3015/#/login')
    page.wait_for_load_state('networkidle')

    # 输入账号
    page.locator('input[type="text"]').fill('11100000000')
    # 输入密码
    page.locator('input[type="password"]').fill('90F11')
    # 点击登录
    page.locator('button[type="submit"]').click()

    # 等待跳转
    page.wait_for_timeout(3000)

    # 获取当前URL
    current_url = page.url
    print(f'当前URL: {current_url}')

    # 截图
    page.screenshot(path='/home/project/test_login_result.png', full_page=True)

    # 检查页面内容
    content = page.content()
    if '标摊客户T' in content or '标摊T有限公司' in content:
        print('登录成功，显示了正确的展商信息')
    elif '账号或密码错误' in content:
        print('登录失败：账号或密码错误')
    else:
        print('页面内容:', content[:800])

    browser.close()

File: test-login.js
const { chromium } = require('playwright');

const testAccounts = [
  { username: 'admin', password: 'admin123', role: '管理员' },
  { username: 'reviewer01', password: 'pwd123', role: '审图员' },
  { username: '17700000000', password: '80F77', role: '标摊展商' },
  { username: '18800000000', password: '80F88', role: '特装展商1' },
  { username: '19900000000', password: '80F99', role: '特装展商2' }
];

async function testLogin() {
  const browser = await chromium.launch({ headless: true });
  const results = [];

  for (const account of testAccounts) {
    const context = await browser.newContext();
    const page = await context.newPage();

    try {
      await page.goto('http://localhost:3015/', { waitUntil: 'networkidle', timeout: 10000 });

      await page.fill('input[type="text"]', account.username);
      await page.fill('input[type="password"]', account.password);
      await page.click('button[type="submit"]');

      await page.waitForTimeout(2000);

      const url = page.url();
      const hasLogout = await page.locator('text=退出登录').count() > 0;

      results.push({
        ...account,
        success: hasLogout,
        currentUrl: url
      });

      await page.click('text=退出登录');
      await page.waitForTimeout(1000);
    } catch (error) {
      results.push({
        ...account,
        success: false,
        error: error.message
      });
    }

    await context.close();
  }

  await browser.close();

  console.log('\n========== 登录测试结果 ==========\n');
  results.forEach((r, i) => {
    const status = r.success ? '✅ 通过' : '❌ 失败';
    console.log(`${i + 1}. ${r.role} (${r.username})`);
    console.log(`   状态: ${status}`);
    if (r.success) {
      console.log(`   跳转URL: ${r.currentUrl}`);
    } else {
      console.log(`   错误: ${r.error}`);
    }
    console.log('');
  });

  const passed = results.filter(r => r.success).length;
  console.log(`总计: ${passed}/${results.length} 个账号测试通过`);
}

testLogin().catch(console.error);
